test/e2e/specs/delete-prohibited/recordset/ind-facet.spec.js

```diff
@@ -0,0 +1,1043 @@
+                                
+                                return chaisePage.clickButton(chaisePage.recordsetPage.getFacetOption(idx, facetParams.option));
+                            }).then(function () {
+                                // wait for request to return
+                                browser.wait(EC.visibilityOf(clearAll), browser.params.defaultTimeout);
+                                
+                                return chaisePage.recordsetPage.getFilters().count();
+                            }).then(function (ct) {
+                                expect(ct).toBe(1, "number of filters is incorrect for '" + facetParams.name + "' facet");
+                                
+                                //should only be one
+                                return chaisePage.recordsetPage.getFilters();
+                            }).then(function (filters) {
+                                return filters[0].getText();
+                            }).then(function(text) {
+                                expect(text).toBe(facetParams.filter, "filter name is inccorect for '" + facetParams.name + "' facet");
+                                
+                                // wait for table rows to load
+                                browser.wait(function () {
+                                    return chaisePage.recordsetPage.getRows().count().then(function(ct) {
+                                        return ct == facetParams.numRows;
+                                    });
+                                }, browser.params.defaultTimeout);
+                                
+                                return chaisePage.recordsetPage.getRows().count();
+                            }).then(function(ct) {
+                                expect(ct).toBe(facetParams.numRows, "number of rows is incorrect for '" + facetParams.name + "' facet");
+                                
+                                return clearAll.click();
+                            }).then(function () {
+                                browser.wait(EC.not(EC.visibilityOf(clearAll)), browser.params.defaultTimeout);
+                                // close the facet
+                                return chaisePage.recordsetPage.getFacetById(idx).click();
+                            }).catch(function (exc) {
+                                console.dir(exc);
+                            });
+                        } else {
+                            //range facet test
+                            if (facetParams.type == "numeric" || facetParams.type == "date" ) {
+                                var minInput, maxInput, minClear, maxClear;
+                                
+                                // open facet
+                                chaisePage.recordsetPage.getFacetById(idx).click().then(function(facet) {
```

Comment:
Is it possible to break this promise change into different `it` statements? It's very difficult to follow what exactly is being tested.